framework/audio/iaudioinsert.h

```diff
@@ -0,0 +1,46 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_AUDIO_IAUDIOINSERT_H
+#define MU_AUDIO_IAUDIOINSERT_H
+
+namespace mu {
+namespace audio {
+class IAudioInsert
+{
+public:
+    virtual ~IAudioInsert() = default;
+
+    virtual void setSampleRate(unsigned int sampleRate) { m_sampleRate = sampleRate; }
```

Comment:
Better not store implementation in header

Reply:
there is not iaudioinsert.cpp file

Reply:
maybe it's better to move it to an abstract class?

Reply:
This class is not like an interface, it must not have  "I" prefix

Reply:
fixed